torch_gauge/o3/rsh.py

```diff
@@ -11,12 +11,13 @@
 import torch
 from joblib import Memory
 from scipy.special import binom, factorial
-
 from torch_gauge import ROOT_DIR
 from torch_gauge.o3.spherical import SphericalTensor
 
 memory = Memory(os.path.join(ROOT_DIR, ".o3_cache"), verbose=0)
 
+def torch_factorial(x):
+  return torch.exp(torch.lgamma(x + 1))
```

Comment:
Just would like to check, does this have the desired behavior at x=1 or x=0?

Reply:
Hey, looks like it did work for x=0 and x=1 but had rounding issues for larger x (eg. x = 9). It turns out the issue with using `scipy.special.factorial` is that it doesn't work on 0 dimensional tensors. So I wrote it like this instead:
```
def torch_factorial(x):
  if x.dim() == 0:
    x = x.unsqueeze(-1)
    out = special.factorial(x)
    return torch.from_numpy(out).squeeze()
  out = special.factorial(x)
  return torch.from_numpy(out)
```
I've tested on zero dimensional tensors (eg. `torch.tensor(9)`) and higher dimensions (eg. `torch.arange(9).reshape(3,3)`). I just comitted this change. I also ran it with one of the example scripts from the OrbNet codebase and it got the same predicted MAE as reported in the paper, so I think everything should be good.